src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -32,11 +36,15 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
         private courseServer: CourseService,
         private translateService: TranslateService,
         private exerciseService: ExerciseService,
+        private accountService: AccountService,
         private route: ActivatedRoute,
     ) {}
 
     ngOnInit() {
         this.exerciseCountMap = new Map<string, number>();
+        this.filters = new Set<number>();
+        this.accountService.identity(true).then(user => (this.user = user));
```

Comment:
I'm not sure if we need the call to the accountService (see next comment), but it would be better to coordinate the asynchronous calls here.
E.g. applyFiltersAndOrder could be called without the user being loaded.

Reply:
True. I removed the call to the service